apps/gp2-server/src/data-providers/reminder.data-provider.ts

```diff
@@ -134,8 +161,98 @@ const getPublishedOutputRemindersFromQuery = (
   );
 };
 
-export const getSortDate = (reminder: gp2Model.ReminderDataObject): DateTime =>
-  DateTime.fromISO(reminder.data.addedDate);
+const getPublishedOutputVersionRemindersFromQuery = (
+  outputVersionCollectionItems: OutputVersionItem[],
+  userProjectIds: string[],
+  userWorkingGroupIds: string[],
+  zone: string,
+): gp2Model.OutputVersionPublishedReminder[] => {
+  if (
+    !outputVersionCollectionItems.length ||
+    (userProjectIds.length === 0 && userWorkingGroupIds.length === 0)
+  ) {
+    return [];
+  }
+
+  outputVersionCollectionItems.sort((reminderA, reminderB) => {
+    const aStartDate = DateTime.fromISO(reminderA.sys.publishedAt);
+    const bStartDate = DateTime.fromISO(reminderB.sys.publishedAt);
+
+    return bStartDate.diff(aStartDate).as('seconds');
+  });
+
+  const seenOutputList: string[] = [];
+
+  return outputVersionCollectionItems.reduce<
+    gp2Model.OutputVersionPublishedReminder[]
+  >((outputVersionReminders, outputVersion) => {
+    const isPublished = !!outputVersion.sys.publishedAt;
+    const output = cleanArray(
+      outputVersion.linkedFrom?.outputsCollection?.items,
+    )[0];
+
+    if (
+      !output ||
+      !output.title ||
+      !output.documentType ||
+      !gp2Model.isOutputDocumentType(output.documentType) ||
+      !isPublished ||
+      !inLast24Hours(outputVersion.sys.publishedAt, zone) ||
+      seenOutputList.includes(output.sys.id)
+    )
+      return outputVersionReminders;
+
+    const { associationName, associationType } =
+      getAssociationNameAndType(output);
+    const userName = getUserName(output);
+
+    const outputEntityIds = (output?.relatedEntitiesCollection?.items || [])
+      .filter((relatedEntityItem) => relatedEntityItem?.sys.id !== undefined)
+      .map((relatedEntityItem) => relatedEntityItem?.sys.id as string);
+
+    const isInProject = outputEntityIds.some((id) =>
+      userProjectIds.includes(id),
+    );
+
+    const isInWorkingGroup = outputEntityIds.some((id) =>
+      userWorkingGroupIds.includes(id),
+    );
+    if (
+      associationName &&
+      associationType &&
+      userName &&
+      ((associationType === 'project' && isInProject) ||
+        (associationType === 'working group' && isInWorkingGroup))
+    ) {
+      seenOutputList.push(output.sys.id);
+      outputVersionReminders.push({
+        id: `output-version-published-${outputVersion.sys.id}`,
+        entity: 'Output Version',
+        type: 'Published',
+        data: {
+          outputId: output.sys.id,
+          documentType: output.documentType,
+          title: output.title,
+          publishedAt: outputVersion.sys.publishedAt,
+          statusChangedBy: userName,
```

Comment:
In the published reminder we can use the name of the person who created the output because since we don't have drafts, the person who created is the one who published it.

But in version reminders, we can't guarantee that the person who created the output is the one who created a version. In CRN it seems that we don't display the name of the person who created the version

<img width="1052" alt="Screenshot 2024-02-08 at 07 48 37" src="https://github.com/yldio/asap-hub/assets/16595804/755cbf4f-2647-4c48-a1ba-9e1e880c2328">

Maybe you could ask Tiff if the name is necessary. Otherwise we would need to add a new field on Outputs to track who was the last person to modify it

Reply:
Discussed this with Tiffany. For now she's agreed to removing the user name as we do in CRN.